File: models/bloodbank/BloodDonor.js
// models/BloodDonor.js
import mongoose from "mongoose";

const BloodDonorSchema = new mongoose.Schema(
  {
    hospitalId: { type: mongoose.Schema.Types.ObjectId, ref: "Hospital", required: true },
    branchId: { type: mongoose.Schema.Types.ObjectId, ref: "Branch", required: true },

    donorId: { type: String, required: true, unique: true },

    firstName: String,
    lastName: String,
    gender: String,
    dob: Date,
    phone: String,
    email: String,
    address: String,

    bloodGroup: { type: mongoose.Schema.Types.ObjectId, ref: "BloodGroup" },

    weight: Number,
    hemoglobin: Number,

    lastDonationDate: Date,
    totalDonations: { type: Number, default: 0 },

    status: {
      type: String,
      enum: ["Active", "Deferred", "Blacklisted"],
      default: "Active",
    },
  },
  { timestamps: true }
);

export default mongoose.model("BloodDonor", BloodDonorSchema);

File: jstojv/src/main/java/com/samrat/jstojv/entity/BloodDonor.java
package com.samrat.jstojv.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "blood_donors")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class BloodDonor extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "hospital_id", nullable = false)
    private Hospital hospital;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "branch_id", nullable = false)
    private Branch branch;

    @Column(name = "donor_id", nullable = false, unique = true)
    private String donorId;

    @Column(name = "first_name")
    private String firstName;
    @Column(name = "last_name")
    private String lastName;
    private String gender;
    private LocalDateTime dob;
    private String phone;
    private String email;
    private String address;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "blood_group_id")
    private BloodGroup bloodGroup;

    private Double weight;
    private Double hemoglobin;

    @Column(name = "last_donation_date")
    private LocalDateTime lastDonationDate;
    @Column(name = "total_donations")
    private Integer totalDonations = 0;

    @Enumerated(EnumType.STRING)
    private Status status = Status.Active;

    public enum Status {
        Active, Deferred, Blacklisted
    }
}
